api/paidAction/boost.js

```diff
@@ -34,12 +34,12 @@ export async function perform ({ invoiceId, sats, id: itemId, ...args }, { me, c
 }
 
 export async function retry ({ invoiceId, newInvoiceId }, { tx, cost }) {
-  await tx.itemAct.updateMany({ where: { invoiceId }, data: { invoiceId: newInvoiceId, invoiceActionState: 'PENDING' } })
   const [{ id, path }] = await tx.$queryRaw`
     SELECT "Item".id, ltree2text(path) as path
     FROM "Item"
     JOIN "ItemAct" ON "Item".id = "ItemAct"."itemId"
-    WHERE "ItemAct"."invoiceId" = ${newInvoiceId}::INTEGER`
+    WHERE "ItemAct"."invoiceId" = ${invoiceId}::INTEGER`
+  await tx.itemAct.updateMany({ where: { invoiceId }, data: { invoiceId: newInvoiceId, invoiceActionState: 'PENDING' } })
```

Comment:
This is just getting the stuff we need using the old `invoiceId` before `tx.itemAct.updateMany()`, since if the action is being converted from invoiced (eg. p2p) to invoiceless (eg. fee credit) the code will unbind the `itemAct` from any invoice and we won't have anything to fetch the data.